rclcpp/include/rclcpp/message_memory_strategy.hpp

```diff
@@ -62,15 +80,52 @@ class MessageMemoryStrategy
     return std::allocate_shared<MessageT, MessageAlloc>(*message_allocator_.get());
   }
 
+  virtual std::shared_ptr<rcl_message_raw_t> borrow_raw_message(unsigned int capacity)
+  {
+    auto raw_msg = std::shared_ptr<rcl_message_raw_t>(new rcl_message_raw_t,
+        [](rmw_message_raw_t * msg) {
+          rmw_raw_message_fini(msg);
+          delete msg;
+        });
+    *raw_msg = rmw_get_zero_initialized_raw_message();
+    rmw_raw_message_init(raw_msg.get(), capacity, &rcutils_allocator_);
```

Comment:
Check the return code, and do not setup the destructor for the `shared_ptr` until you know this succeeds. Make sure to throw based on the rcl ret code.